src/app/jscaip/tests/RulesUtils.spec.ts

```diff
@@ -1,47 +1,50 @@
-import { LegalityStatus } from '../LegalityStatus';
 import { MGPNode } from '../MGPNode';
 import { Minimax } from '../Minimax';
 import { Move } from '../Move';
 import { Player } from '../Player';
 import { GameStatus, Rules } from '../Rules';
-import { AbstractGameState } from '../GameState';
-import { comparableEquals, ComparableObject } from 'src/app/utils/Comparable';
+import { GameState } from '../GameState';
+import { comparableEquals, isComparableObject } from 'src/app/utils/Comparable';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+import { MGPFallible } from 'src/app/utils/MGPFallible';
 export class RulesUtils {
 
-    public static expectMoveSuccess(rules: Rules<Move, AbstractGameState>,
-                                    state: AbstractGameState,
-                                    move: Move,
-                                    expectedState: AbstractGameState)
+    public static expectMoveSuccess<R extends Rules<M, S, L>, M extends Move, S extends GameState, L>(
+        rules: R,
+        state: S,
+        move: M,
+        expectedState: S)
     : void
     {
-        const legality: LegalityStatus = rules.isLegal(move, state);
-        expect(legality.legal).toBeTruthy();
```

Comment:
reason of the illegality should be the context of some assert !

Reply:
it is? le else ajoute legality.getReason()